src/RetroMole.Core/Utility/IEnumExtensions.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Collections;
+
+namespace RetroMole.Core.Utility;
+
+public static class IEnumerableExtensions
+{
+    public static IEnumerable<TSource> ApplyIf<TSource>(
+        this IEnumerable<TSource> enumerable,
+        bool condition,
+        Func<TSource, TSource> apply
+        ) => condition
+            ? enumerable.Apply(apply)
+            : enumerable;
+    public static IEnumerable<TSource> Apply<TSource>(
+        this IEnumerable<TSource> enumerable,
+        Func<TSource, TSource> apply
+    ) => enumerable.Select(e => apply(e));
+}
```

Comment:
This is just `Select` without a return type but okay ig
Do add a nl at the end of the file tho